compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5056,75 +5085,66 @@ private STNode parseCheckingKeyword() {
     }
 
     /**
-     * Parse fail expression. This method is used to parse both fail expression
-     * as well as fail action.
-     *
-     * <p>
-     * <code>
-     * fail-expr := fail-keyword expression
-     * fail-action := fail-keyword action
-     * </code>
+     * Parse continue statement.
+     * <code>continue-stmt := continue ; </code>
      *
-     * @param allowActions Allow actions
-     * @param isRhsExpr    Is rhs expression
-     * @return Fail expression node
+     * @return continue statement
      */
-    private STNode parseFailExpression(boolean isRhsExpr, boolean allowActions, boolean isInConditionalExpr) {
-        STNode failKeyword = parseFailKeyword();
-        STNode expr =
-                parseExpression(OperatorPrecedence.EXPRESSION_ACTION, isRhsExpr, allowActions, isInConditionalExpr);
-        if (isAction(expr)) {
-            return STNodeFactory.createFailExpressionNode(SyntaxKind.FAIL_ACTION, failKeyword, expr);
-        } else {
-            return STNodeFactory.createFailExpressionNode(SyntaxKind.FAIL_EXPRESSION, failKeyword, expr);
-        }
+    private STNode parseContinueStatement() {
+        startContext(ParserRuleContext.CONTINUE_STATEMENT);
+        STNode continueKeyword = parseContinueKeyword();
+        STNode semicolon = parseSemicolon();
+        endContext();
+        return STNodeFactory.createContinueStatementNode(continueKeyword, semicolon);
     }
 
     /**
-     * Parse fail keyword.
-     * <p>
-     * <code>
-     * fail-keyword := fail
-     * </code>
+     * Parse continue-keyword.
      *
-     * @return Parsed node
+     * @return continue-keyword node
      */
-    private STNode parseFailKeyword() {
+    private STNode parseContinueKeyword() {
         STToken token = peek();
-        if (token.kind == SyntaxKind.FAIL_KEYWORD) {
+        if (token.kind == SyntaxKind.CONTINUE_KEYWORD) {
             return consume();
         } else {
-            recover(token, ParserRuleContext.FAIL_KEYWORD);
-            return parseFailKeyword();
+            recover(token, ParserRuleContext.CONTINUE_KEYWORD);
+            return parseContinueKeyword();
         }
     }
 
     /**
-     * Parse continue statement.
-     * <code>continue-stmt := continue ; </code>
+     * Parse fail statement.
+     * <code>fail-stmt := return [ action-or-expr ] ;</code>
```

Comment:
grammar rule seems to be incorrect

Reply:
Fixed with 3189bc6c6a9ee435ca57fa863f10d8e4f357ab6b

Reply:
Fixed with 3189bc6c6a9ee435ca57fa863f10d8e4f357ab6b